### PR TITLE
Counter metric implementation

### PR DESCRIPTION
**Pull request by Llandy3d**
This includes the `Counter` metric that will be useful for experimenting.
It also includes a `LockedValue` backend, as a stepping stone for working on pluggable backends.

### CODE REVIEW

File: pytheus/metrics.py [hunk 1]
```diff
@@ -111,13 +120,38 @@ def __init__(
...
+        """
+        Will count and reraise raised exceptions.
+        It is possibly to specify which exceptions to track.
+        """
+        if exceptions is None:
+            exceptions = Exception
+
+        try:
+            yield
+        except exceptions:
```
Comment (Llandy3d) [inline pytheus/metrics.py:152]: mypy complains here with `error: Exception type must be derived from BaseException` but I've ignored it for now

File: tests/test_metrics.py [hunk 2]
```diff
@@ -50,3 +50,43 @@ def test_validate_label_with_incorrect_values(self, label):
...
+
+    def test_count_exception(self, counter):
+        with pytest.raises(ValueError):
+            with counter.count_exceptions():
+                raise ValueError
+
+        assert counter._value.get() == 1
+
+    def test_count_exception_with_specified(self, counter):
+        with pytest.raises(ValueError):
```
Comment (Llandy3d) [inline tests/test_metrics.py:81]: this is testing that the exception will actually be raised again

File: pytheus/backends.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from threading import Lock
+
+
+class LockedValue:
```
Comment (Llandy3d) [inline pytheus/backends.py:4]: for experimenting I created only this one, but work will be needed on the pluggable system

**Review by d14c — APPROVED** [review]
LGTM 👍

**Pull request merged**